lnbits/core/views/api.py

```diff
@@ -364,13 +368,8 @@ async def api_payments_sse(
 
 
 @core_app.get("/api/v1/payments/{payment_hash}")
-async def api_payment(payment_hash, X_Api_Key: Optional[str] = Header(None)):
-    wallet = None
-    try:
-        if X_Api_Key.extra:
-            print("No key")
-    except:
-        wallet = await get_wallet_for_key(X_Api_Key)
+async def api_payment(payment_hash, wal: WalletTypeInfo = Depends(require_invoice_key)):
+    wallet = wal.wallet
```

Comment:
@arcbtc: why not use this? It seems to work according to my tests.

Reply:
Because it doesn't require a key. You can find out if an invoice is paid or not without a key, or add a key and get all the data back.
ie `curl -X GET https://legend.lnbits.com/api/v1/payments/afb06542db0671c78bc1d16c8e8380c8f1a5238bec74755cd4f91bfd3a39d1a1 `
vs
`curl -X GET https://legend.lnbits.com/api/v1/payments/afb06542db0671c78bc1d16c8e8380c8f1a5238bec74755cd4f91bfd3a39d1a1 -H "X-Api-Key: 8a5e8e06d37540d6b945a0f0593d00f9" -H "Content-type: application/json"`

Reply:
Gotcha! I've returned it back to its original state with a bit a cleaner code. Is it ok now?